File: src/main/java/com/qi/qismorecommands/command/StringMaker.java
// StringMaker.java
package com.qi.qismorecommands.command;

import java.util.Random;

public class StringMaker {
    private static final Random RANDOM = new Random();

    // 合并字符串
    public static String joinStrings(String str1, String str2) {
        return str1 + str2;
    }

    // 转换为大写
    public static String toUpperCase(String str) {
        return str.toUpperCase();
    }

    // 转换为小写
    public static String toLowerCase(String str) {
        return str.toLowerCase();
    }

    // 生成随机字母字符串
    public static String randomLetters(int length) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < length; i++) {
            char c = (char) (RANDOM.nextBoolean() ?
                    RANDOM.nextInt(26) + 'a' : RANDOM.nextInt(26) + 'A');
            sb.append(c);
        }
        return sb.toString();
    }

    // 生成随机数字字符串
    public static String randomNumbers(int length) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < length; i++) {
            sb.append(RANDOM.nextInt(10));
        }
        return sb.toString();
    }

    // 生成随机混合字符串
    public static String randomMixed(int length) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < length; i++) {
            int type = RANDOM.nextInt(3);
            if (type == 0) {
                sb.append((char) (RANDOM.nextInt(26) + 'a'));
            } else if (type == 1) {
                sb.append((char) (RANDOM.nextInt(26) + 'A'));
            } else {
                sb.append(RANDOM.nextInt(10));
            }
        }
        return sb.toString();
    }

    // 生成自定义字符集随机字符串
    public static String randomChars(String charset, int length) {
        if (charset == null || charset.isEmpty()) {
            return "";
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < length; i++) {
            int index = RANDOM.nextInt(charset.length());
            sb.append(charset.charAt(index));
        }
        return sb.toString();
    }

    // 阿拉伯数字转简单中文数字 (12345 → 一二三四五)
    public static String numberToChineseSimple(String numberStr) {
        if (numberStr == null || numberStr.isEmpty()) {
            return "";
        }
        char[] digits = numberStr.toCharArray();
        StringBuilder result = new StringBuilder();
        for (char digit : digits) {
            if (Character.isDigit(digit)) {
                result.append("零一二三四五六七八九".charAt(digit - '0'));
            } else {
                result.append(digit); // 保留非数字字符
            }
        }
        return result.toString();
    }

    // 阿拉伯数字转大写中文数字 (12345 → 壹贰叁肆伍)
    public static String numberToChineseCapital(String numberStr) {
        if (numberStr == null || numberStr.isEmpty()) {
            return "";
        }
        char[] digits = numberStr.toCharArray();
        StringBuilder result = new StringBuilder();
        for (char digit : digits) {
            if (Character.isDigit(digit)) {
                result.append("零壹贰叁肆伍陆柒捌玖".charAt(digit - '0'));
            } else {
                result.append(digit); // 保留非数字字符
            }
        }
        return result.toString();
    }

    // 阿拉伯数字转标准中文数字 (12345 → 一万二千三百四十五)
    public static String numberToChineseNormal(String numberStr) {
        try {
            long number = Long.parseLong(numberStr);
            return convertWithUnits(number, false);
        } catch (NumberFormatException e) {
            return "无效数字";
        }
    }

    // 阿拉伯数字转财务中文数字 (12345 → 壹万贰仟叁佰肆拾伍)
    public static String numberToChineseFinancial(String numberStr) {
        try {
            long number = Long.parseLong(numberStr);
            return convertWithUnits(number, true);
        } catch (NumberFormatException e) {
            return "无效数字";
        }
    }

    // 带单位的数字转换核心方法
    private static String convertWithUnits(long number, boolean financial) {
        if (number == 0) {
            return "零";
        }

        // 定义单位和数字
        String[] units = financial ?
                new String[]{"", "拾", "佰", "仟", "万", "亿", "兆", "京"} :
                new String[]{"", "十", "百", "千", "万", "亿", "兆", "京"};

        String[] digits = financial ?
                new String[]{"零", "壹", "贰", "叁", "肆", "伍", "陆", "柒", "捌", "玖"} :
                new String[]{"零", "一", "二", "三", "四", "五", "六", "七", "八", "九"};

        // 处理负数
        if (number < 0) {
            return ("负") + convertWithUnits(-number, financial);
        }

        StringBuilder result = new StringBuilder();
        long[] unitValues = {1, 10, 100, 1000, 10000, 100000000L, 1000000000000L, 10000000000000000L};

        // 从最大单位开始处理
        for (int i = unitValues.length - 1; i >= 0; i--) {
            if (number >= unitValues[i]) {
                long quotient = number / unitValues[i];
                if (quotient > 0) {
                    if (quotient < 10) {
                        // 直接添加数字和单位
                        result.append(digits[(int) quotient]).append(units[i]);
                    } else {
                        // 递归处理更大的数字
                        result.append(convertWithUnits(quotient, financial)).append(units[i]);
                    }
                }
                number %= unitValues[i];

                // 添加零（如果需要）
                if (number > 0 && number < unitValues[i] / 10) {
                    result.append(digits[0]);
                }
            }
        }

        // 处理十位数的一十 → 十
        String resultStr = result.toString();
        if (resultStr.startsWith("一十")) {
            resultStr = resultStr.substring(1);
        } else if (resultStr.startsWith("壹拾")) {
            resultStr = resultStr.substring(1);
        }

        return resultStr;
    }
}

File: src/main/kotlin/com/qi/qismorecommands/command/StringVar/StringMaker.kt
package com.qi.qismorecommands.command.StringVar

import java.util.Locale
import kotlin.random.Random

object StringMaker {
    // 合并字符串
    fun joinStrings(str1: String?, str2: String): String {
        return (str1 ?: "") + str2
    }

    // 转换为大写
    fun toUpperCase(str: String): String {
        return str.uppercase(Locale.getDefault())
    }

    // 转换为小写
    fun toLowerCase(str: String): String {
        return str.lowercase(Locale.getDefault())
    }

    // 生成随机字母字符串
    fun randomLetters(length: Int): String {
        return buildString {
            repeat(length) {
                val charCode = if (Random.Default.nextBoolean()) {
                    Random.Default.nextInt(26) + 'a'.code
                } else {
                    Random.Default.nextInt(26) + 'A'.code
                }
                append(charCode.toChar())
            }
        }
    }

    // 生成随机数字字符串
    fun randomNumbers(length: Int): String {
        return buildString {
            repeat(length) {
                append(Random.Default.nextInt(10))
            }
        }
    }

    // 生成随机混合字符串
    fun randomMixed(length: Int): String {
        return buildString {
            repeat(length) {
                when (Random.Default.nextInt(3)) {
                    0 -> append((Random.Default.nextInt(26) + 'a'.code).toChar())
                    1 -> append((Random.Default.nextInt(26) + 'A'.code).toChar())
                    else -> append(Random.Default.nextInt(10))
                }
            }
        }
    }

    // 生成自定义字符集随机字符串
    fun randomChars(charset: String?, length: Int): String {
        if (charset.isNullOrEmpty()) return ""

        return buildString {
            repeat(length) {
                val index = Random.Default.nextInt(charset.length)
                append(charset[index])
            }
        }
    }

    // 阿拉伯数字转简单中文数字 (12345 → 一二三四五)
    fun numberToChineseSimple(numberStr: String?): String {
        if (numberStr.isNullOrEmpty()) return ""

        return buildString {
            numberStr.forEach { char ->
                if (char.isDigit()) {
                    append("零一二三四五六七八九"[char.digitToInt()])
                } else {
                    append(char)
                }
            }
        }
    }

    // 阿拉伯数字转大写中文数字 (12345 → 壹贰叁肆伍)
    fun numberToChineseCapital(numberStr: String?): String {
        if (numberStr.isNullOrEmpty()) return ""

        return buildString {
            numberStr.forEach { char ->
                if (char.isDigit()) {
                    append("零壹贰叁肆伍陆柒捌玖"[char.digitToInt()])
                } else {
                    append(char)
                }
            }
        }
    }

    // 阿拉伯数字转标准中文数字 (12345 → 一万二千三百四十五)
    fun numberToChineseNormal(numberStr: String): String {
        return numberStr.toLongOrNull()?.let { number ->
            convertWithUnits(number, false)
        } ?: "无效数字"
    }

    // 阿拉伯数字转财务中文数字 (12345 → 壹万贰仟叁佰肆拾伍)
    fun numberToChineseFinancial(numberStr: String): String {
        return numberStr.toLongOrNull()?.let { number ->
            convertWithUnits(number, true)
        } ?: "无效数字"
    }

    // 带单位的数字转换核心方法
    private fun convertWithUnits(number: Long, financial: Boolean): String {
        if (number == 0L) return "零"
        if (number < 0) return "负" + convertWithUnits(-number, financial)

        val units = if (financial) {
            arrayOf("", "拾", "佰", "仟", "万", "亿", "兆", "京")
        } else {
            arrayOf("", "十", "百", "千", "万", "亿", "兆", "京")
        }

        val digits = if (financial) {
            arrayOf("零", "壹", "贰", "叁", "肆", "伍", "陆", "柒", "捌", "玖")
        } else {
            arrayOf("零", "一", "二", "三", "四", "五", "六", "七", "八", "九")
        }

        val unitValues = longArrayOf(1, 10, 100, 1000, 10000, 100000000L, 1000000000000L, 10000000000000000L)
        var remaining = number
        val result = StringBuilder()

        for (i in unitValues.indices.reversed()) {
            if (remaining >= unitValues[i]) {
                val quotient = remaining / unitValues[i]
                if (quotient > 0) {
                    if (quotient < 10) {
                        result.append(digits[quotient.toInt()]).append(units[i])
                    } else {
                        result.append(convertWithUnits(quotient, financial)).append(units[i])
                    }
                }
                remaining %= unitValues[i]

                if (remaining > 0 && remaining < unitValues[i] / 10) {
                    result.append(digits[0])
                }
            }
        }

        val resultStr = result.toString()
        return when {
            resultStr.startsWith("一十") -> resultStr.substring(1)
            resultStr.startsWith("壹拾") -> resultStr.substring(1)
            else -> resultStr
        }
    }
}